linux/power_operation_scripts/set_power_script.yml

```diff
@@ -0,0 +1,111 @@
+# Copyright 2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+#   Set and check soft power operation
+# # Parameters:
+#   power_script_op: power script operation
+#   power_script_type: default or custom
+
+- name: "Start to test {{ power_script_type }} {{ power_script_op }} script"
+  ansible.builtin.debug:
+    msg:
+      - "Start to test {{ power_script_type }} {{ power_script_op }} script"
+
+- name: "Set {{ power_script_type }} script for {{ power_script_op }} script"
+  include: set_power_script_{{ power_script_type }}.yml
+  vars:
+    power_script_op: "{{ power_script_op }}"
+
+- name: "Check the current configuration for {{ power_script_op }} script"
+  ansible.builtin.command: "{{ vmware_toolbox_cmd_path }} script {{ power_script_op }} current"
+  register: power_script_result
+  delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Print the current configuration for {{ power_script_op }} script"
+  ansible.builtin.debug: var=power_script_result.stdout
+
+- name: "Check current configuration is expected"
+  ansible.builtin.assert:
+    that:
+      - power_script_result.stdout is defined
+      - power_script_result.stdout.strip() == script_file_path
+    fail_msg: "The {{ power_script_ops }} script is {{ power_script_result.stdout | default('') }}, not as expected {{ script_file_path }}."
+    success_msg: "The {{ power_script_ops }} script is {{ power_script_result.stdout | default('') }}, which is as expected."
+
+- name: "Get content of file tools.conf"
+  ansible.builtin.command: "cat /etc/vmware-tools/tools.conf"
+  register: tools_conf_content_result
+  delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Check the configuration is expected in tools.conf"
+  ansible.builtin.assert:
+    that:
+      - tools_conf_content_result.stdout is defined
+      - tools_conf_content_result.stdout.find(power_script_file) != -1
```

Comment:
power_script_file -> script_file_in_conf_file
And you need to add a condition `- script_file_in_conf_file` to check it is not empty before using it